charts/kube-prometheus-stack/values.yaml

```diff
@@ -3303,6 +3303,42 @@ prometheus:
       #   hosts:
       #     - prometheus.example.com
 
+  # -- BETA: Configure the gateway routes for the chart here.
+  # More routes can be added by adding a dictionary key like the 'main' route.
+  # Be aware that this is an early beta of this feature,
+  # kube-prometheus-stack does not guarantee this works and is subject to change.
+  # Being BETA this can/will change in the future without notice, do not use unless you want to take that risk
+  # [[ref]](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io%2fv1alpha2)
+  route:
+    main:
+      # -- Enables or disables the route
+      enabled: false
+
+      # -- Set the route apiVersion, e.g. gateway.networking.k8s.io/v1 or gateway.networking.k8s.io/v1alpha2
+      apiVersion: gateway.networking.k8s.io/v1
```

Comment:
I'm not sure this should be configurable. `v1alpha2` is already deprecated, IMHO there is there is value in supporting `v1` only.

Reply:
Gateway api is too new to enforce only v1 api versions imo, there may be many implementations lacking behind, which would mean those users are unable to use this feature in the chart. There are also some update issues iirc, with regards to experimental versions of v1 api, which might cause users to still be on alpha api versions.